### PR TITLE
Bind table/config API to lazy fetch table config.

### PR DESCRIPTION
**Pull request by maxhutch**
# Citrine Python PR

## Description 
There's an API that gets the config from a GEM Table.  This binds
that lazily, such that we only GET the config when it is referenced.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

**Comment by sdedkins** [comment]
> @kroenlein @sdedkins thoughts?

Looks great to me. Is /definition a new route or does that already exist?

**Comment by maxhutch** [comment]
> Looks great to me. Is /definition a new route or does that already exist?

It already exists.  @kroenlein and I chatted last Friday and I have some adjustments to make.  We can definitely make it a property so you don't need the `()`.

File: src/citrine/resources/gemtables.py [hunk 1]
```diff
@@ -41,12 +41,35 @@ class GemTable(Resource['Table']):
     """:str: URL pointing to the location of the GEM Table's contents.
     This is an expiring download link and is not unique."""
 
+    _config = properties.Optional(properties.Object(TableConfig), "config")
```
Comment (maxhutch) [inline src/citrine/resources/gemtables.py:44]: By defining these as optional properties, we'll seamlessly start using them if they're added to the API responses.

**Review by kroenlein — APPROVED** [review]
Questions, but no issues

File: src/citrine/resources/gemtables.py [hunk 2]
```diff
@@ -41,12 +41,35 @@ class GemTable(Resource['Table']):
...
+        if self._name is None:
+            self._name = self.config.name
+        return self._name
+
+    @property
+    def description(self) -> str:
+        if self._description is None:
+            self._description = self.config.description
+        return self._description
+
```
[the remark this answers is not in the document]
Comment (maxhutch) [reply src/citrine/resources/gemtables.py:72]: They don't need this special logic. You can already access them, e.g. `table.version`.

File: src/citrine/resources/table_config.py [hunk 3]
```diff
@@ -397,6 +397,12 @@ def get(self, uid: Union[UUID, str], *, version: Optional[int] = None):
             data['version'] = data['versions'][index]
         return self.build(data)
 
+    def get_for_table(self, table: "GemTable") -> TableConfig:
```
[the remark this answers is not in the document]
Comment (maxhutch) [reply src/citrine/resources/table_config.py:400]: I tried with no luck.  Will try again as I clean up the code.

File: src/citrine/resources/gemtables.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kroenlein) [reply src/citrine/resources/gemtables.py]: I asked if they should be readonly.  For example,
```
table.uid = '844e0221-8c4d-403a-bfba-b0ed44777c51'
```
changes the value.  Not blocking.

**Review by kroenlein — APPROVED** [review]

**Pull request merged**